dubbo-config/dubbo-config-api/src/test/java/org/apache/dubbo/integration/single/SingleRegistryCenterDubboProtocolIntegrationTest.java

```diff
@@ -234,17 +251,38 @@ private void afterExport() {
         // MetadataInfo's service exists or not
         Assertions.assertNotNull(serviceInfo);
         // The name of MetadataInfo's service is right or not
-        Assertions.assertEquals(serviceInfo.getName(),SingleRegistryCenterIntegrationService.class.getName());
+        Assertions.assertEquals(serviceInfo.getName(), SingleRegistryCenterIntegrationService.class.getName());
         // The group of MetadataInfo's service is right or not
         Assertions.assertNull(serviceInfo.getGroup());
         // The version of MetadataInfo's service is right or not
         Assertions.assertNull(serviceInfo.getVersion());
         // The protocol of MetadataInfo's service is right or not
-        Assertions.assertEquals(serviceInfo.getProtocol(),PROTOCOL_NAME);
+        Assertions.assertEquals(serviceInfo.getProtocol(), PROTOCOL_NAME);
         // The serviceKey of MetadataInfo's service is right or not
-        Assertions.assertEquals(serviceInfo.getServiceKey(),SingleRegistryCenterIntegrationService.class.getName());
+        Assertions.assertEquals(serviceInfo.getServiceKey(), SingleRegistryCenterIntegrationService.class.getName());
         // The matchKey of MetadataInfo's service is right or not
-        Assertions.assertEquals(serviceInfo.getMatchKey(),key);
+        Assertions.assertEquals(serviceInfo.getMatchKey(), key);
+        // The exported services are right or not
+        // 1. The exported service must contain SingleRegistryCenterIntegrationService and MetadataService
+        // 2. The exported service's interface must be SingleRegistryCenterIntegrationService.class
+        // 3. All exported service must be exported
+        exportedServiceListener = (ExportedServiceListener) ExtensionLoader.getExtensionLoader(ServiceListener.class).getExtension("exported");
+        Assertions.assertNotNull(exportedServiceListener);
+        Assertions.assertEquals(exportedServiceListener.getExportedServices().size(), 2);
```

Comment:
Here add the logic to check exportedService is 1 after singleRegistryCenterServiceConfig.unexport.

Reply:
It's really a good testcase for exportedService. However, this will affect the following testcases of consumer side if we do that.

I have already designed the related testcase for checking exportedService after unexport

```java
public void tearDown() throws IOException {
    DubboBootstrap.reset();
    PROVIDER_APPLICATION_NAME = null;
    PROTOCOL_NAME = null;
    PROTOCOL_PORT = 0;
    serviceConfig = null;
    referenceConfig = null;
    // The exported service has been unexported
    Assertions.assertTrue(exportedServiceListener.getExportedServices().isEmpty());
    exportedServiceListener = null;
    logger.info(getClass().getSimpleName() + " testcase is ending...");
    // destroy zookeeper only once
    logger.info(SingleZooKeeperServer.getZookeeperServerName() + " is beginning to shutdown...");
    SingleZooKeeperServer.shutdown();
    logger.info(SingleZooKeeperServer.getZookeeperServerName() + " has shutdown.");
}
```

Reply:
fine.